### PR TITLE
ROX-13249 plop: use falco as process info source

### PR DESCRIPTION
**Pull request by ovalenti**
## Description

Using /proc as the only source of process information exposes us to cmdline being modified by the process (as is the case for nginx). Here, we replace our ProcessStore::ISource by the falco library, which keeps information as seen in the fork syscall.

## Checklist
- [ ] Investigated and inspected CI test results
- ~~[ ] Updated documentation accordingly~~

**Automated testing**
  - ~~[ ] Added unit tests~~
  - ~~[ ] Added integration tests~~ Those are already in place and show the issues
  - ~~[ ] Added regression tests~~

### CODE REVIEW

**Comment by robbycochran** [comment]
`SysdigService::ByPID` will be executed for every connection event but only utilized when `data.state == TCP_LISTEN`. Did you consider the performance impact? What about lazily getting process details from Falco when a listen endpoint is encountered?

File: collector/lib/SysdigService.h [hunk 1]
```diff
@@ -89,6 +93,7 @@ class SysdigService : public Sysdig {
 
   void AddSignalHandler(std::unique_ptr<SignalHandler> signal_handler);
 
+  mutable std::mutex sysdig_mutex_;
```
Comment (robbycochran) [inline collector/lib/SysdigService.h:96]: The Falco project has endeavored to remove `sysdig` from the codebase after donation to the CNCF, please rename this to remove `sysdig`, something like `libsinsp_mutex` maybe?

**Comment by ovalenti** [comment]
> `SysdigService::ByPID` will be executed for every connection event but only utilized when `data.state == TCP_LISTEN`. Did you consider the performance impact? What about lazily getting process details from Falco when a listen endpoint is encountered?

@robbycochran, I blindly replaced PID by process references and missed the performance impact. Thanks for spotting this.

The current code uses PID until we know that we need process information, and does the lookup from there.

File: collector/lib/SysdigService.h (hunk 1 above)
Comment (ovalenti) [reply collector/lib/SysdigService.h]: Renamed as suggested.

Do you think that we should rename the whole SysdigService as well ?

**Review by Molter73 — COMMENTED** [review]
I'll put more general comments on #900

**Review by Stringy — COMMENTED** [review]
Changes generally look good! wrt the performance impact, do we have any way of measuring the difference? I'll soon re-introduce the benchmarking back into CI (via ansible) and that may help, but I feel we need more targeted and isolated measurements

**Pull request closed without merging**